### ISSUE TITLE
Problems loading MDX files when converting to typescript

### ISSUE THREAD

**Issue by ethancrook99** [storybookjs/storybook#12241]
**Describe the bug**
When converting an existing project and changing a `*.stories.js` file to a `*.stories.tsx` file, any mdx pages that were previously used in the story fail to load.

**To Reproduce**

1. Add a typescript configuration to a project that uses storybook
2. Change a `*.stories.js` file to use the `.tsx` extension
3. Run storybook and any previously used mdx files loaded by that story shouldn't show up

**Expected behavior**
When converting a story to typescript, the behavior of MDX files should remain the same as when using plain javascript.

**Code snippets**

```
// Icons.mdx (shortened)

import { Preview, Story, Props } from '@storybook/addon-docs/blocks'
import { BaseIcon, ToggleIcon, DirectionalIcon } from './BaseIcon'
import { StatusBanner } from '../../storybook-helpers'

<StatusBanner packageName="zion-icon" />

# Zion Icon
<Props of={ToggleIcon} />
```

```
// Icons.stories.js

import page from './Icons.mdx'

export default {
  title: 'UI Foundation/Icons',
  component: [],
  parameters: { docs: { page } },
}
```

**System:**
```
Environment Info:
  System:
    OS: macOS 10.15.6
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  Binaries:
    Node: 12.16.3 - ~/.nvm/versions/node/v12.16.3/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.4 - /usr/local/bin/npm
  Browsers:
    Chrome: 84.0.4147.135
    Safari: 13.1.2
  npmPackages:
    @storybook/addon-a11y: ^5.3.13 => 5.3.19 
    @storybook/addon-actions: ^5.3.13 => 5.3.19 
    @storybook/addon-console: ^1.2.1 => 1.2.1 
    @storybook/addon-docs: ^5.3.13 => 5.3.19 
    @storybook/addon-knobs: ^5.3.13 => 5.3.19 
    @storybook/addon-links: ^5.3.18 => 5.3.19 
    @storybook/addon-storysource: ^5.3.13 => 5.3.19 
    @storybook/addon-viewport: ^5.3.13 => 5.3.19 
    @storybook/addons: ^5.3.13 => 5.3.19 
    @storybook/components: ^5.3.13 => 5.3.19 
    @storybook/preset-typescript: ^3.0.0 => 3.0.0 
    @storybook/react: ^5.3.13 => 5.3.19 
    @storybook/theming: ^5.3.13 => 5.3.19 
```

**Additional context**
The project I'm working in is a monorepo, and all the components that use typescript are functioning as expected outside of storybook.

**Comment by shilman** [comment]
Try upgrading to 6.0?

```
npx sb upgrade
```

[Full migration guide](https://medium.com/storybookjs/storybook-6-migration-guide-200346241bb5)

**Comment by danielstocks** [comment]
@shilman I'm having issues with the `sb upgrade` as advertised in the migration guide.

```
$ npx sb upgrade
 Invalid command: Command {
  commands: [],
  options: [],
  _execs: Set {},
  _allowUnknownOption: false,
  _args: [],
  _name: '*',
  _optionValues: {},
  _storeOptionsAsProperties: true,
  _passCommandToAction: true,
  _actionResults: [],
  _helpFlags: '-h, --help',
  _helpDescription: 'output usage information',
  _helpShortFlag: '-h',
  _helpLongFlag: '--help',
  _noHelp: true,
  _exitCallback: undefined,
  _executableFile: undefined,
  parent: [Command]
}.
 See --help for a list of available commands.
str.toLowerCase is not a function
```

Running `sb info` reveals the following
```
$ npx sb
Usage: sb <command> [options]

Options:
  -V, --version                  output the version number
  -h, --help                     output usage information

Commands:
  init [options]                 Initialize Storybook into your project.
  add [options] <addon>          Add an addon to your Storybook
  info                           Prints debugging information about the local environment
  migrate [options] [migration]  Run a storybook codemod migration on your source files
```

There is no `upgrade` command (only migrate), or am I missing something?

**Comment by shilman** [comment]
@danielstocks `upgrade` was added in 6.0. perhaps you have an older version of `sb` installed on your dev machine? what happens when you run `npx sb@latest upgrade`?

**Comment by danielstocks** [comment]
@shilman Thanks that worked! Might be worth mentioning in the migration guide as it's far from obvious why it doesn't work if you coming from storybook 5!

**Comment by ethancrook99** [comment]
@shilman I've upgraded to 6.0, and I'm still running into the same problem. Any ideas on what it could be? I'm happy to provide more information if there's anything you need to know

**Comment by shilman** [comment]
@ethancrook99 what's the error?

**Comment by ethancrook99** [comment]
@shilman I'm getting `Cannot find module './Icons.mdx' or its corresponding type declarations.`, but I'm not sure how to get around that without using `// @ts-ignore` (which gets rid of the error but the mdx file still doesn't load properly).

**Comment by shilman** [comment]
@ethancrook99 Do you have a repro repo you can share?

**Comment by ethancrook99** [comment]
@shilman I don't, but I think I found the source of the issue. I made a pretty big oversight--I left out `ts` and `tsx` from the regex that adds stories in my `preview.js` (I've updated it before, but evidently I didn't commit the changes). So it went from `require.context('../packages/icon/src', true, /\.stories\.(js|mdx)$/)` to `require.context('../packages/icon/src', true, /\.stories\.(js|ts|tsx|mdx)$/)`. It looks like that fixed the problem. 

I also noticed that when I run filtered storybook to only show one package, it creates the same error in `/.storybookfiltered` by not adding `ts` and `tsx` to the require context. I'll look into that one further and see if there are any config or script changes I need to make to get that working correctly.

Thank you so much for all your help!
